lib/openstack/amqp/openstack_qpid_connection.rb

```diff
@@ -71,9 +71,24 @@ def connection # :nodoc:
   end
 
   def create_connection
-    url = "#{@options[:hostname]}:#{@options[:port]}"
-    opts = @options.select { |k, v| [:username, :password].include? k }
-    opts[:transport] = "ssl" if @options[:port] == 5671
-    Qpid::Messaging::Connection.new(:url => url, :options => opts)
+    Qpid::Messaging::Connection.new(:url => connection_url, :options => connection_options)
+  end
+
+  def connection_options
+    connection_opts = {}
+    if @options[:port] == "5671"
+      connection_opts[:transport]     = "ssl"
+      connection_opts[:ssl_cert_name] = @options[:hostname]
+    end
+    if @options.key?(:username)
+      connection_opts[:sasl_mechanism] = "PLAIN"
+      connection_opts[:username]       = @options[:username]
+      connection_opts[:password]       = @options[:password]
+    end
+    connection_opts
+  end
+
+  def connection_url
+    "#{@options[:hostname]}:#{@options[:port]}"
```

Comment:
@blomquisg 
Depends... what is qpid using it for?

If it's a URI, it needs to be square bracketed around the hostname.
It's hard to tell because the scheme is missing... http, https?
See: https://github.com/ManageIQ/manageiq/blob/8cceedf6d142d725084a006eeb8a9a3d48f020d7/lib/VMwareWebService/MiqVimClientBase.rb#L39

Something like this(http):
```
      uri = URI::HTTP.build(:port => @options[:port])
      uri.hostname = @options[:hostname]
      uri.to_s
```